File: algorithms/sorting/insertionsort1/insertionsort1-java/insertionsort1-java/src/TesteInsertionSort1.java
import static org.junit.Assert.assertEquals;

import java.util.*;
import org.junit.Test;

public class TesteInsertionSort1 {

    @Test
    public void testeCaso1() {

        int n = 5;
        List<Integer> arr = Arrays.asList(2, 4, 6, 8, 3);
        List<Integer> resultado = Resultado.insertionSort1(n, arr);
        List<Integer> esperado = Arrays.asList(2, 3, 4, 6, 8);

        assertEquals(resultado, esperado);
    }
    
}

File: algorithms/sorting/insertionsort1/insertionsort1-javascript/js/testeInsertionSort1.js
import {insertionSort1} from '../js/insertionSort1.js';
import assert from 'assert';

describe('insertionSort1', function() {

    it('deve retornar um array ordenado em ordem não decrescente [2, 3, 4, 6, 8]', function() {

        let n = 5;
        let arr = [2, 4, 6, 8, 3];
        let esperado = [2, 3, 4, 6, 8];

        let resultado = insertionSort1(n, arr);

        assert.deepStrictEqual(resultado, esperado);
    });
});